WD-Firmware/comrv/comrv.c

```diff
@@ -18,34 +18,777 @@
+   }
+
+   /* get actual function/data offset */
+   usOffset = M_COMRV_GET_TOKEN_OFFSET_IN_BYTES(unToken);
+
+   /* are we returning to an overlay function */
+   if (ucIsInvoke == D_COMRV_RET_CALLER_BIT_0)
+   {
+      /* calculate the function return offset; at this point
+         pComrvStackFrame->uiCalleeToken holds the return address */
+#ifdef D_COMRV_MULTI_GROUP_SUPPORT
+      /* In multi group token we need to get the offset from previously loaded token */
+      if (pComrvStackFrame->usCalleeMultiGroupTableEntry)
+      {
+         /* we now are at the point of loading a multi-group toke so we need to take the
+            previous token for which the return address refers to */
+         unToken = pOverlayMultiGroupTokensTable[pComrvStackFrame->usCalleeMultiGroupTableEntry];
+         /* get the offset */
+         uiTemp = M_COMRV_GET_TOKEN_OFFSET_IN_BYTES(unToken);
+         /* get the token group size */
+         usOverlayGroupSize = M_COMRV_GET_OVL_GROUP_SIZE_IN_BYTES(unToken);
+         /* calculate the actual return offset */
+         usOffset += ((u32_t)(pComrvStackFrame->uiCalleeToken) - uiTemp) & (--usOverlayGroupSize);
+      }
+      else
+      {
+#endif /* D_COMRV_MULTI_GROUP_SUPPORT */
+         /* calculate the actual return offset */
+         usOffset += ((u32_t)(pComrvStackFrame->uiCalleeToken) - usOffset) & (--usOverlayGroupSize);
+#ifdef D_COMRV_MULTI_GROUP_SUPPORT
+      }
+#endif /* D_COMRV_MULTI_GROUP_SUPPORT */
+   }
+   /* we are calling a new overlay function or accessing overlay data */
+   else
+   {
+#ifdef D_COMRV_MULTI_GROUP_SUPPORT
+     /* Update comrv stack frame with the loaded multi group table entry.
+        It is used to calculate the actual return offset in case we
+        are returning to a multi group token */
+      pComrvStackFrame->usCalleeMultiGroupTableEntry = usSelectedMultiGroupEntry;
+#endif /* D_COMRV_MULTI_GROUP_SUPPORT */
+   }
+
+#ifdef D_COMRV_FW_INSTRUMENTATION
+   comrvNotificationHook(uiProfilingIndication , unToken.ucValue);
+#endif /* D_COMRV_FW_INSTRUMENTATION */
+
+   /* group is now loaded to memory so we can return the address of the data/function */
+   return (void*)((u08_t*)pAddress + usOffset);
+}
+
+/**
+* Get comrv heap eviction candidates according to a given size
+*
+* @param requestedEvictionSize - size requested for eviction; expressed in
+*                                units of D_COMRV_OVL_GROUP_SIZE_MIN
+*        pEvictCandidatesList  - output eviction candidate list of comrv heap indexes
+* @return number of eviction candidates in the output list
+*/
+u08_t comrvGetEvictionCandidates(u08_t ucRequestedEvictionSize, u08_t* pEvictCandidatesList)
+{
+   u08_t ucAccumulatedSize = 0, ucIndex = 0;
+   u08_t ucEntryIndex, ucNumberOfCandidates = 0;
+   u32_t uiEvictCandidateMap[D_COMRV_EVICT_CANDIDATE_MAP_SIZE], uiCandidates;
+
+   /* first lets clear the uiCandidates list */
+   memset(uiEvictCandidateMap, 0, sizeof(u32_t)*D_COMRV_EVICT_CANDIDATE_MAP_SIZE);
```

Comment:
"D_COMRV_EVICT_CANDIDATE_MAP_SIZE" link this define to the configuration data
replace "memset" with "comrv memset"

Reply:
done in e1b829c